src/swaps/fastbtc/FastBTCWithdrawSwapProvider.ts

```diff
@@ -0,0 +1,394 @@
+  blockNumber: number; // the RSK block number where this was initialized
+  totalAmountSatoshi: number; // the number of BTC satoshis that the user sent
+  btcAddress: string; // the BTC address in legacy or Bech32 encoded format
+}
+
+class FastBTCWithdrawSwapProvider extends SwapProvider {
+  config: FastBtcWithdrawSwapProviderConfig;
+  _provider: JsonRpcProvider;
+
+  constructor(config: FastBtcWithdrawSwapProviderConfig) {
+    super(config);
+    const rpcUrl = buildConfig.rskRpcUrls[this.config.network];
+    const chainId = ChainNetworks[ChainId.Rootstock][this.config.network].chainId;
+    this._provider = new ethers.providers.StaticJsonRpcProvider(rpcUrl, chainId);
+  }
+
+  getFastBtcBridge(provider: JsonRpcProvider) {
+    const fastBtcBridge = new ethers.Contract(this.config.routerAddress, ABI, provider);
+    return fastBtcBridge;
+  }
+
+  async getLimits() {
+    const fastBtcBridge = this.getFastBtcBridge(this._provider);
+    const minInSatoshi = await fastBtcBridge.minTransferSatoshi();
+    const maxInSatosih = await fastBtcBridge.maxTransferSatoshi();
+    const min = unitToCurrency(cryptoassets.BTC, minInSatoshi);
+    const max = unitToCurrency(cryptoassets.BTC, maxInSatosih);
+
+    return { min, max };
+  }
+
+  public getClient(network: Network, walletId: string, asset: string, accountId: string) {
+    return super.getClient(network, walletId, asset, accountId) as Client<EvmChainProvider, EvmWalletProvider>;
+  }
+
+  public getReceiveClient(network: Network, walletId: string, asset: string, accountId: string) {
+    return super.getClient(network, walletId, asset, accountId) as Client<
+      BitcoinEsploraApiProvider,
+      BitcoinBaseWalletProvider
+    >;
+  }
+
+  async getSupportedPairs() {
+    const { min, max } = await this.getLimits();
+    return [
+      {
+        from: 'RBTC',
+        to: 'BTC',
+        rate: '0.998',
+        min: min.toFixed(),
+        max: max.toFixed(),
+      },
+    ];
+  }
+
+  async getQuote(quoteRequest: QuoteRequest) {
+    const { from, to, amount } = quoteRequest;
+    if (from !== 'RBTC' || to !== 'BTC') {
+      return null;
+    }
+
+    const fastBtcBridge = this.getFastBtcBridge(this._provider);
+    const fromAmountInSatoshi = new BN(currencyToUnit(cryptoassets.BTC, new BN(amount)));
+    const feeSatoshi: ethers.BigNumber = await fastBtcBridge.calculateCurrentFeeSatoshi(
+      ethers.BigNumber.from(fromAmountInSatoshi.toFixed())
+    );
+    const fromAmountInUnit = currencyToUnit(cryptoassets[from], new BN(amount));
+    const toAmountInUnit = fromAmountInSatoshi.minus(new BN(feeSatoshi.toString()));
+
+    return {
+      // TODO: Why is from amount being set in every quote call? It should be in one place only (getQuotes action)
+      fromAmount: fromAmountInUnit.toFixed(),
+      toAmount: toAmountInUnit.toFixed(),
+    };
+  }
+
+  async buildSwapTx({ network, walletId, quote }: { network: Network; walletId: WalletId; quote: SwapQuote }) {
```

Comment:
[Pebble] I think it would be better to avoid inline type declaration for ease of reuse.

Reply:
fixed